### PR TITLE
Update chain work and checkpoint for 2.0.4

### LINKED ISSUE #3886: Update chain work and checkpoint for 2.0.4

### PR DESCRIPTION
**Pull request by bitcartel**
Closes #3886

### CODE REVIEW

**Comment by bitcartel** [comment]
For testing updates to zkbot: @zkbot try

**Comment by bitcartel** [comment]
There was a zkbot try to test CI, which passed on key builders, but then failed due to low disk space.  Does not impact review.

File: src/chainparams.cpp [hunk 1]
```diff
@@ -181,11 +181,12 @@ class CMainParams : public CChainParams {
             (222222, uint256S("0x000000000cafb9e56445a6cabc8057b57ee6fcc709e7adbfa195e5c7fac61343"))
             (270000, uint256S("0x00000000025c1cfa0258e33ab050aaa9338a3d4aaa3eb41defefc887779a9729"))
             (304600, uint256S("0x00000000028324e022a45014c4a4dc51e95d41e6bceb6ad554c5b65d5cea3ea5"))
-            (410100, uint256S("0x0000000002c565958f783a24a4ac17cde898ff525e75ed9baf66861b0b9fcada")),
-            1539405939,     // * UNIX timestamp of last checkpoint block
-            3954156,        // * total number of transactions between genesis and last checkpoint
+            (410100, uint256S("0x0000000002c565958f783a24a4ac17cde898ff525e75ed9baf66861b0b9fcada"))
+            (497000, uint256S("0x0000000000abd333f0acca6ffdf78a167699686d6a7d25c33fca5f295061ffff")),
+            1552501838,     // * UNIX timestamp of last checkpoint block
+            4463933,        // * total number of transactions between genesis and last checkpoint
```
Comment (str4d) [inline src/chainparams.cpp]: I confirmed that this matches what my local `zcashd` node sees.

File: src/chainparams.cpp [hunk 2]
```diff
@@ -181,11 +181,12 @@ class CMainParams : public CChainParams {
...
                             //   (the tx=... number in the SetBestChain debug.log lines)
-            5553            // * estimated number of transactions per day after checkpoint
+            5173            // * estimated number of transactions per day after checkpoint
```
Comment (str4d) [inline src/chainparams.cpp]: Checks out.

File: src/chainparams.cpp [hunk 3]
```diff
@@ -108,7 +108,7 @@ class CMainParams : public CChainParams {
         consensus.vUpgrades[Consensus::UPGRADE_SAPLING].nActivationHeight = 419200;
 
         // The best chain should have at least this much work.
-        consensus.nMinimumChainWork = uint256S("0x00000000000000000000000000000000000000000000000000a95cc5099213e3");
+        consensus.nMinimumChainWork = uint256S("0x00000000000000000000000000000000000000000000000000c12875ded911cf");
```
Comment (str4d) [inline src/chainparams.cpp]: I checked this with my local `zcashd` node.

**Comment by bitcartel** [comment]
@zkbot r+

**Comment by bitcartel** [comment]
Unrelated error on Ubuntu 16 builder:
```
MiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:11117 using version 170003
MiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:11117 using version 170006
MiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:11117 using version 170002
Unexpected exception caught during testing: NodeConn instance has no attribute 'rejectMessage'
Stopping nodes
Cleaning up
Failed
!!! FAIL: p2p_nu_peer_management.py !!!
```

**Comment by bitcartel** [comment]
@zkbot retry

**Pull request merged**